src/SignalR/common/SignalR.Common/src/Protocol/HandshakeProtocol.cs

```diff
@@ -123,59 +106,51 @@ public static bool TryParseResponseMessage(ref ReadOnlySequence<byte> buffer, ou
                 return false;
             }
 
-            var textReader = Utf8BufferTextReader.Get(payload);
+            var reader = new Utf8JsonReader(in payload, isFinalBlock: true, state: default);
+
+            reader.CheckRead();
+            reader.EnsureObjectStart();
 
-            try
+            int? minorVersion = null;
+            string error = null;
+
+            while (reader.CheckRead())
             {
-                using (var reader = JsonUtils.CreateJsonTextReader(textReader))
+                if (reader.TokenType == JsonTokenType.PropertyName)
                 {
-                    JsonUtils.CheckRead(reader);
-                    JsonUtils.EnsureObjectStart(reader);
-
-                    int? minorVersion = null;
-                    string error = null;
+                    var memberName = reader.ValueSpan;
```

Comment:
ValueSpan won't always have the token (when your input payload is within a multi-segment `ReadOnlySequence`, check `HasValueSequence` and call `ValueSequence.ToArray()` for tokens that straddle more than one segment

Reply:
>This doesn't matter since we pass in a byte[] correct?

Doesn't look like to me. It is coming from pipe `ReadResult`:
https://github.com/aspnet/AspNetCore/blob/3cd5054eb5d8851b25881cd05cccb931aaa022e6/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L806-L817

TryParseMessage can return a multi-segment ReadOnlySequence:
https://github.com/aspnet/AspNetCore/blob/51be998c3cfe4258eecf30bd01207e3288de51d9/src/SignalR/common/Shared/TextMessageParser.cs#L13

Which will eventually get passed into this `Utf8JsonReader`:
https://github.com/aspnet/AspNetCore/blob/51be998c3cfe4258eecf30bd01207e3288de51d9/src/SignalR/common/SignalR.Common/src/Protocol/HandshakeProtocol.cs#L109